paddle/fluid/framework/data_set.h

```diff
@@ -168,6 +176,8 @@ class DatasetImpl : public Dataset {
   // and when finish reading, we set cur_channel = 1 - cur_channel,
   // so if cur_channel=0, all data are in output_channel, else consume_channel
   int cur_channel_;
+  std::vector<T> original_data_;
```

Comment:
maybe  shuffle_slot_original_data_  to make it clear

Reply:
fixed